[Bug]: @storybook/addon-themes v9.0.4 has undeclared dependency on React

### Describe the bug

I am developing a Svelte app, been using Storybook 8 fine with it.

Upgraded storybook from v8 to v9, only packages I have after upgrade are:

```
"@storybook/addon-themes": "^9.0.4",
"@storybook/svelte-vite": "^9.0.4",
"storybook": "^9.0.4",
```

Now getting the following error:

```
› yarn storybook
storybook v9.0.4

info => Starting manager..
info => Starting preview..
9:54:56 am [vite] (client) Re-optimizing dependencies because lockfile has changed
╭────────────────────────────────────────────────────╮
│                                                    │
│   Storybook 9.0.4 for svelte-vite started          │
│   134 ms for manager and 497 ms for preview        │
│                                                    │
│    Local:            http://localhost:6006/        │
│    On your network:  http://192.168.1.109:6006/    │
│                                                    │
╰────────────────────────────────────────────────────╯
✘ [ERROR] Could not resolve "react"

    node_modules/@storybook/addon-themes/dist/index.mjs:4:18:
      4 │ import React from 'react';
        ╵                   ~~~~~~~
```

@storybook/addon-themes relies on React, but does not declare it as a dependency:

https://github.com/storybookjs/storybook/blob/next/code/addons/themes/package.json

Running `yarn add -D react` fixes the issue, but I dont want to declare react as a dependency in my app.

### Reproduction link

TODO

### Reproduction steps

_No response_

### System

```bash
System:
    OS: macOS 15.5
    CPU: (12) arm64 Apple M3 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.15.1 - ~/.asdf/installs/nodejs/22.15.1/bin/node
    Yarn: 4.9.1 - /opt/homebrew/bin/yarn <----- active
    npm: 10.9.2 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 137.0.7151.56
    Safari: 18.5
  npmPackages:
    @storybook/addon-themes: ^9.0.4 => 9.0.4 
    @storybook/svelte-vite: ^9.0.4 => 9.0.4 
    storybook: ^9.0.4 => 9.0.4
```

### Additional context

_No response_

--- a/code/addons/themes/package.json
+++ b/code/addons/themes/package.json
@@ -70,6 +70,8 @@
   },
   "devDependencies": {
     "@storybook/icons": "^1.4.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
     "typescript": "^5.8.3"
   },
   "peerDependencies": {

